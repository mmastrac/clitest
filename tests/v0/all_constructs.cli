#!/usr/bin/env clitest --version=0

#   - # <comment>
#   - $ <command>
#   - ? <grok pattern>
#   - ! <literal>
#   - repeat { ... }
#   - option { ... }
#   - choice { ... }
#   - unordered { ... }
#   - sequence { ... }
#   - ignore { ... }
#   - reject { ... }

$ echo "a\nb\nc"
repeat {
    choice {
        ! a
        ! b
        ! c
    }
}

$ echo "a\nb\nc"
unordered {
    ! c
    ! a
    ! b
}

$ echo "a\nb\nc"
sequence {
    ! a
    sequence {
        ! b
        sequence {
            ! c
        }
    }
}

$ echo "a\nb\nc"
! a
optional {
    ! doesn't happen!!
}
! b
! c

$ echo "a\nWARNING: UH OH\nb\nc"
ignore {
    ? WARNING: %{DATA}
}
!!!
a
b
c
!!!
