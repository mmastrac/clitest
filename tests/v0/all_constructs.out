Running <root>/all_constructs.cli ...

printf "a\nb\nc\n"
---
a
b
c
---
✅ OK

printf "a\nb\nc\n"
---
a
b
c
---
✅ OK

printf "a\nb\nc\n"
---
a
b
c
---
✅ OK

printf "a\nb\nc\n"
---
a
b
c
---
✅ OK

printf "a\nWARNING: UH OH\nb\nc\n"
---
a
WARNING: UH OH
b
c
---
✅ OK

printf "hello\nworld\n"
---
hello
world
---
✅ OK

<root>/all_constructs.cli PASSED
